pgmpy/tests/test_factors/test_Factor.py

```diff
@@ -670,3 +670,8 @@ def test_cardinality(self):
     def tearDown(self):
         del self.rule_cpd_without_rules
 
+    def test_to_tree_cpd(self):
+        tree_cpd_string = repr(self.rule_cpd_with_rules.to_tree_cpd())
+        self.assertEqual(tree_cpd_string, '<pgmpy.factors.CPD.TreeCPD object at {address}'
+                         .format(address=tree_cpd_string.split(" ")[-1]))
+
```

Comment:
@siteshjaiswal Add tests for checking if correct edges were created or not, all the nodes are present or not etc.